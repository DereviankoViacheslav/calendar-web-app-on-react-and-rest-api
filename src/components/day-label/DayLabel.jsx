import React from 'react';
import './DayLabel.scss';
import moment from 'moment';
import PropTypes from 'prop-types';

class DayLabel extends React.PureComponent {
  render() {
    let nextDay = moment(this.props.date).format('YYYY-MM-DD');
    const labels = [...new Array(7)].map(() => {
      const today = moment().format('YYYY-MM-DD');
      const todayStyle = moment(today).isSame(nextDay) ? ' today' : '';
      const labelElem = (
        <div key={nextDay} className="label-day">
          <span className="label-day__title">{moment(nextDay).format('ddd')}</span>
          <div className={`label-day__date${todayStyle}`}>{moment(nextDay).format('DD')}</div>
          <div className="decor-line"></div>
        </div>);
      nextDay = moment(nextDay).add('days', 1).format('YYYY-MM-DD');
      return labelElem;
    });
    
    return (
      <div className="labels">
        {labels}
      </div>
    );
  }
}

DayLabel.propTypes = {
  date: PropTypes.object.isRequired,
}

export default DayLabel;