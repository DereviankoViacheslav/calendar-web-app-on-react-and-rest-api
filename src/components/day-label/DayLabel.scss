.labels {
  position: fixed;
  width: 100%;
  display: flex;
  top: 80px;
  z-index: 99;
  padding-left: 60px;
  border-bottom: 1px solid #d5d5d5;
  background-color: white;
}

.label-day {
  position: relative;
  flex-grow: 1;
  height: 90px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;

  &__title {
    font-size: 18px;
    opacity: 0.5;
  }

  &__date {
    width: 40px;
    height: 40px;
    align-self: center;
    margin-top: 5px;
    font-size: 22px;
    text-align: center;
    line-height: 40px;
    border-radius: 50%;
  }
}

.decor-line {
  position: absolute;
  bottom: 0;
  height: 20px;
  width: 100%;
  border-left: 1px solid #d5d5d5;
}

.today {
  background-color: royalblue;
  color: white;
}