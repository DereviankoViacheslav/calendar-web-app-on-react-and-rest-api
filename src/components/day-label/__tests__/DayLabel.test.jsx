import React from 'react';
import { shallow } from 'enzyme';
import DayLabel from '../DayLabel'
import moment from 'moment';

describe('<DayLabel />', () => {
  const props = {
    date: moment('2020-02-12'),
  };

  it('Get snapshot', () => {
    const wrappedComponent = shallow(<DayLabel {...props} />);
    expect(wrappedComponent).toMatchSnapshot();
  });

  it('should render labels', () => {
    const wrappedComponent = shallow(<DayLabel {...props} />);
    expect(wrappedComponent.find('.labels').exists()).toBeTruthy();
  });
});