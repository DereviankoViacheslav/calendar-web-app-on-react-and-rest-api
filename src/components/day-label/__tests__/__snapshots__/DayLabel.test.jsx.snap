// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DayLabel /> Get snapshot 1`] = `
<div
  className="labels"
>
  <div
    className="label-day"
    key="2020-02-12"
  >
    <span
      className="label-day__title"
    >
      Wed
    </span>
    <div
      className="label-day__date"
    >
      12
    </div>
    <div
      className="decor-line"
    />
  </div>
  <div
    className="label-day"
    key="2020-02-13"
  >
    <span
      className="label-day__title"
    >
      Thu
    </span>
    <div
      className="label-day__date"
    >
      13
    </div>
    <div
      className="decor-line"
    />
  </div>
  <div
    className="label-day"
    key="2020-02-14"
  >
    <span
      className="label-day__title"
    >
      Fri
    </span>
    <div
      className="label-day__date"
    >
      14
    </div>
    <div
      className="decor-line"
    />
  </div>
  <div
    className="label-day"
    key="2020-02-15"
  >
    <span
      className="label-day__title"
    >
      Sat
    </span>
    <div
      className="label-day__date today"
    >
      15
    </div>
    <div
      className="decor-line"
    />
  </div>
  <div
    className="label-day"
    key="2020-02-16"
  >
    <span
      className="label-day__title"
    >
      Sun
    </span>
    <div
      className="label-day__date"
    >
      16
    </div>
    <div
      className="decor-line"
    />
  </div>
  <div
    className="label-day"
    key="2020-02-17"
  >
    <span
      className="label-day__title"
    >
      Mon
    </span>
    <div
      className="label-day__date"
    >
      17
    </div>
    <div
      className="decor-line"
    />
  </div>
  <div
    className="label-day"
    key="2020-02-18"
  >
    <span
      className="label-day__title"
    >
      Tue
    </span>
    <div
      className="label-day__date"
    >
      18
    </div>
    <div
      className="decor-line"
    />
  </div>
</div>
`;
